### PR TITLE
Initial setup of amazon-cloudwatch-observability helm chart

### PR DESCRIPTION
**Pull request by sky333999**
*Issue #, if available:*

*Description of changes:* Initial setup of amazon-cloudwatch-observability helm chart

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: helm/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+{{- default "default" .Values.manager.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "cloudwatch-agent.serviceAccountName" -}}
+{{- if .Values.agent.enabled }}
+{{- default (include "cloudwatch-agent.name" .) .Values.agent.serviceAccount.name }}
```
Comment (nathalapooja) [inline helm/templates/_helpers.tpl:50]: The service account name expected is "amazon-cloudwatch-agent-operator-agent" does this create that?

File: helm/templates/admission-webhooks/operator-webhook-with-cert-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,222 @@
+{{- if and (.Values.admissionWebhooks.create) (.Values.admissionWebhooks.certManager.enabled) }}
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: {{ printf "%s/%s-serving-cert" .Release.Namespace (include "amazon-cloudwatch-observability.name" .) }}
```
Comment (nathalapooja) [inline helm/templates/admission-webhooks/operator-webhook-with-cert-manager.yaml:6]: is this -serving-cert correct cert name ?

**Review by nathalapooja — APPROVED** [review]

File: helm/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+{{/*
+Common labels
+*/}}
+{{- define "amazon-cloudwatch-observability.labels" -}}
+{{ include "amazon-cloudwatch-observability.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: EKS
```
Comment (lisguo) [inline helm/templates/_helpers.tpl:23]: Is the EKS value constant? Customers can install on native k8s, will it also have this label?

File: helm/values.yaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      repository: public.ecr.aws/aws-observability/aws-for-fluent-bit
+      tag: 2.31.12.20230911
+
+## Provide CloudWatchAgent Operator manager container image and resources.
+##
+manager:
+  name:
+  image:
+    # This repository is for internal beta testing only, it will be replaced with the public repository before public release
+    repository: 506463145083.dkr.ecr.us-west-2.amazonaws.com/cwagent-operator-release
```
Comment (lisguo) [inline helm/values.yaml:33]: can we change these to use the public ecr repo?

File: helm/values.yaml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    repository: public.ecr.aws/cloudwatch-agent/cloudwatch-agent
+    tag: 1.300028.1b210
+  enabled: true
+  serviceAccount:
+    name: # override agent service account name 
+  config: # optional config that can be provided to override the defaultConfig
+  defaultConfig:
+    {
+      "agent": {
+        "debug": true
```
Comment (lisguo) [inline helm/values.yaml:125]: we should remove this

File: helm/values.yaml [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      },
+      "traces": {
+        "traces_collected": {
+          "apm": {
```
Comment (lisguo) [inline helm/values.yaml:129]: and this

File: helm/values.yaml [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+          }
+        }
+      },
+      "logs": {
+        "metrics_collected": {
+          "apm": {
```
Comment (lisguo) [inline helm/values.yaml:135]: and this

File: helm/values.yaml [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+manager:
+  name:
+  image:
+    # This repository is for internal beta testing only, it will be replaced with the public repository before public release
+    repository: 506463145083.dkr.ecr.us-west-2.amazonaws.com/cwagent-operator-release
+    tag: latest
+  autoInstrumentationImage:
+    java:
+      # This repository is for internal beta testing only, it will be replaced with the public repository before public release
+      repository: 160148376629.dkr.ecr.us-west-2.amazonaws.com/aws-apm-preview
```
Comment (lisguo) [inline helm/values.yaml:38]: we should reference the adot public repo for now

File: helm/templates/_helpers.tpl (hunk 3 above)
Comment (sky333999) [reply helm/templates/_helpers.tpl:23]: Yeah we will update this before we release a public helm chart. For now, the helm chart is specific for the EKS add-on.

File: helm/templates/_helpers.tpl (hunk 1 above)
Comment (sky333999) [reply helm/templates/_helpers.tpl:50]: We are updating the naming of the components for the ones associated with CWA to remain consistent with historical naming. So this is an intentional change to not call it that anymore.

File: helm/templates/admission-webhooks/operator-webhook-with-cert-manager.yaml (hunk 2 above)
Comment (sky333999) [reply helm/templates/admission-webhooks/operator-webhook-with-cert-manager.yaml:6]: Yeah this mainly needs to match the name of the certificate itself [here](https://github.com/aws/amazon-cloudwatch-agent-operator/pull/29/files#diff-43bd35744a58a4155b7a13ec827ab3a8660fa7d8e1154123f11a508e01505c5bR11)

File: helm/values.yaml (hunk 4 above)
Comment (sky333999) [reply helm/values.yaml:33]: Yup let me update these now

File: helm/values.yaml (hunk 8 above)
Comment (sky333999) [reply helm/values.yaml:38]: Yup let me update these now

File: helm/values.yaml (hunk 5 above)
Comment (sky333999) [reply helm/values.yaml:125]: Yeah I was going to remove these before the final add-on release on Wed

**Review by nathalapooja — APPROVED** [review]

**Review by lisguo — APPROVED** [review]

**Pull request merged**